src/coreclr/tools/Common/TypeSystem/Ecma/EcmaModule.cs

```diff
@@ -585,7 +585,7 @@ private Object ResolveAssemblyReference(AssemblyReferenceHandle handle)
             an.CultureName = _metadataReader.GetString(assemblyReference.Culture);
             an.ContentType = GetContentTypeFromAssemblyFlags(assemblyReference.Flags);
 
-            var assembly = _moduleResolver.ResolveAssembly(an, throwIfNotFound: false);
+            var assembly = _moduleResolver.ResolveAssembly(an);
             if (assembly == null)
                 return ResolutionFailure.GetAssemblyResolutionFailure(an.Name);
```

Comment:
the original code seems correct. is it another layer that is not properly handling the returned value from ResolutionFailure.GetAssemblyResolutionFailure?

Reply:
> ResolutionFailure.GetAssemblyResolutionFailure

I assume you're referring to the call immediately after the changed line?  It never reaches it, which is why the original code is incorrect.  `ILVerifyTypeSystemContext.ResolveAssembly` calls `ILVerifyTypeSystemContext.ResolveAssemblyOrNetmodule`, which executes the following code:

```
            PEReader peReader = _resolver.Resolve(simpleName); //returns null
            if (peReader == null && throwIfNotFound) //throwIfNotFound is false
            {
                throw new VerifierException("Assembly or module not found: " + simpleName);
            }
            var module = GetModule(peReader, containingAssembly); //returns null
            VerifyModuleName(verificationName, module); //blows up because it expects module to be not null
            return module;
```

If we assume that the original code is correct, and the expected behavior is to return null here, then should a null check be added to the top of `VerifyModuleName` instead?

Reply:
I think it would be better to return null as soon as we find that this is not going to work:
```
            if (peReader == null)
            {
                if (throwIfNotFound)
                    throw new VerifierException("Assembly or module not found: " + simpleName);
                return null;
            }
```
I think the early out for null can be deleted from `GetModule` with this fix.

Reply:
> I think the early out for null can be deleted from `GetModule` with this fix.

It can't yet.  `GetModule` is called from multiple places.  One of them is `Program.VerifyAssembly`, and there is no null check in place there.

Reply:
> I think it would be better to return null as soon as we find that this is not going to work

Done.  Thanks for the feedback!